toolchains/workspace-pnpm/run_in_dir.py

```diff
@@ -3,16 +3,23 @@
 Runs a program in a directory.
 """
 import argparse
+import json
 import os
 import subprocess
 import sys
 
-def compute_path(arg: str) -> str:
-    if arg.endswith("::abspath"):
-        return os.path.abspath(arg.removesuffix("::abspath"))
+def merge_env_from_file(file_path):
+    # Shell command to source the .env file and then get the environment as JSON using jq
+    if file_path and os.path.exists(file_path):
+        cmd = f'source {file_path} && jq -n env'
     else:
-        return arg
+        cmd = f'jq -n env'
```

Comment:
I would prefer if we could do this in python only...

Reply:
Done here: https://github.com/GaloyMoney/galoy/pull/3372/commits/6eac5741fcc11e4ccee8336156a06fae7083fbf0